web/src/attachments_ui.js

```diff
@@ -66,6 +67,44 @@ function delete_attachments(attachment, file_name) {
     });
 }
 
+export function suggest_delete_detached_attachments(attachments_list) {
+    const html_body = render_confirm_delete_detached_attachments_modal({
+        attachments_list,
+    });
+    dialog_widget.launch({
+        id: "confirm_delete_attachments_modal",
+        html_heading: $t_html({defaultMessage: "Delete uploaded files?"}),
+        html_body,
+        html_submit_button: $t_html({defaultMessage: "Delete"}),
+        html_exit_button: $t_html({defaultMessage: "Don't delete"}),
+        on_click() {
+            const requests = attachments_list.map(({id}) =>
+                channel.async_request(channel.del, "/json/attachments/" + id),
```

Comment:
Can you talk about why you need this `async_request` mechanism -- is the idea to avoid closing the modal until these have all completed? I guess that makes sense given this UI, probably?

I kinda wonder whether it'd be better UI to have each have its own "Delete" button, so you can delete some but not all if you so choose, and a "delete all" option.

Reply:
OK I confirmed with @alya that this UI is fine so let's stick with it.